frontend/src/modules/cruise-applications/pages/ApplicationsPage.tsx

```diff
@@ -1,68 +1,131 @@
+import { AppAvatar } from '@/core/components/AppAvatar';
+import { AppButton } from '@/core/components/AppButton';
+import { AppBadge } from '@/core/components/AppBadge';
 
 export function ApplicationsPage() {
   const applicationsQuery = useCruiseApplicationsQuery();
 
   const columns: ColumnDef<CruiseApplicationDto>[] = [
     {
-      header: 'Numer/data',
-      accessorFn: (row) => `${row.number}/${row.date}`,
+      id: 'number',
+      header: 'Numer',
+      accessorFn: (row) => `${row.number}`,
+      sortDescFirst: true,
     },
     {
+      id: 'date',
+      header: 'Data',
+      accessorFn: (row) => `${row.date}`,
+    },
+    {
+      id: 'year',
       header: 'Rok rejsu',
       accessorFn: (row) => `${row.year}`,
     },
     {
+      id: 'avatar',
+      header: undefined,
+      accessorFn: (row) => `${row.cruiseManagerFirstName} ${row.cruiseManagerLastName}`,
+      cell: (cell) => <AppAvatar fullName={cell.getValue() as string} variant="small" />,
+      enableColumnFilter: false,
+      enableSorting: false,
+      size: 40,
+    },
+    {
+      id: 'manager',
       header: 'Kierownik',
       accessorFn: (row) => `${row.cruiseManagerFirstName} ${row.cruiseManagerLastName}`,
     },
     {
+      id: 'forms',
       header: 'Formularze',
-      cell: ({ row }) => (
+      cell: ({ row }) => {
+        const isFormBReadOnly = row.original.status === CruiseApplicationStatus.Accepted;
+        return (
         <div className="flex flex-col gap-1">
           <AppLink disabled={!row.original.hasFormA} href={`/applications/${row.original.id}/formA`}>
             Formularz A
           </AppLink>
-          <AppLink disabled={!row.original.hasFormB} href={`/applications/${row.original.id}/formB`}>
+          <AppLink disabled={!row.original.hasFormB} href={`/applications/${row.original.id}/formB?mode=${isFormBReadOnly ? 'view' : 'preview'}`}>
             Formularz B
           </AppLink>
           <AppLink disabled={!row.original.hasFormC} href={`/applications/${row.original.id}/formC`}>
             Formularz C
           </AppLink>
         </div>
-      ),
+      );},
     },
     {
+      id: 'points',
       header: 'Punkty',
-      accessorFn: (row) => `${row.points}`,
+      accessorFn: (row) => `${row.points} pkt.`,
+      cell: ({ row }) => <AppBadge>{row.original.points} pkt.</AppBadge>,
     },
     {
+      id: 'status',
       header: 'Status',
       accessorFn: (row) => `${row.status}`,
+      cell: ({ row }) => (
+        <>
+          <p className="italic">{row.original.status}</p>
+          {row.original.status === CruiseApplicationStatus.FormBRequired && (
+            <AppButton size="plain" type="link" href={`/cruises/${row.original.id}/formB?mode=edit`}>
+              Wypełnij
+            </AppButton>
+          )}
+          {row.original.status === CruiseApplicationStatus.Undertaken && (
+            <>
```

Comment:
I think it would be cleaner to have `div` instead of `<>`. Then apply the correct flex styles so it's clear how the resulting html will look like